http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -415,6 +422,51 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       _ <- gceRuntimeMonitor.process(runtime.id).compile.drain
     } yield ()
 
+  private[monitor] def handleCreateDiskMessage(msg: CreateDiskMessage, now: Instant)(
+    implicit traceId: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
```

Comment:
I'm picturing a flow like the following here (psuedocode):
```
val createDisk = for {
  googleResp <- googleService.createDisk(…)
  _ <- // update persistent disk in the DB with async fields like `googleId`
  _ <- // monitor disk creation
} yield ()

createDisk.handleErrorWith { 
  // if an error occurred, log and set the disk status to `Error`
}
```

See `handleCreateRuntimeMessage` as a model. I think monitoring stuff is tracked as a separate ticket (and will change once https://github.com/DataBiosphere/leonardo/pull/1395/ is merged).

Reply:
I'm also wondering if we want to make Google calls inline here in the pubsub subscriber, or create a separate algebra/interpreter similar to `RuntimeAlgebra` and `Gce/DataprocInterpreter`. I'm not sure I see the need for a separate algebra yet, so maybe let's keep things simple for now and just make the calls here in `LeoPubsubMessageSubscriber`.